testsuite/diem-backend/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use async_trait::async_trait;
+
+pub mod k8s_node;
+pub mod k8s_swarm;
+pub mod local_node;
+pub mod local_swarm;
+
+use diem_crypto::test_utils::KeyPair;
+use diem_crypto::ed25519::{Ed25519PrivateKey, Ed25519PublicKey};
+use diem_types::waypoint::Waypoint;
+use reqwest::{Client, Url};
+use diem_types::chain_id::ChainId;
+use std::fmt;
+use std::str::FromStr;
+use diem_client::Client as JsonRpcClient;
+
+#[derive(Clone)]
+enum InstanceBackend {
+    Swarm,
+    K8s,
+}
+
+#[derive(Clone)]
+pub struct Cluster {
+    validator_instances: Vec<Instance>,
+    fullnode_instances: Vec<Instance>,
+    lsr_instances: Vec<Instance>,
+    vault_instances: Vec<Instance>,
+    mint_key_pair: KeyPair<Ed25519PrivateKey, Ed25519PublicKey>,
+    waypoint: Option<Waypoint>,
+    pub chain_id: ChainId,
+}
+
+impl Cluster {
+    pub fn validator_instances(&self) -> &[Instance] {
+        &self.validator_instances
+    }
+
+    pub fn fullnode_instances(&self) -> &[Instance] {
+        &self.fullnode_instances
+    }
+
+    pub fn lsr_instances(&self) -> &[Instance] {
+        &self.lsr_instances
+    }
+
+    pub fn vault_instances(&self) -> &[Instance] {
+        &self.vault_instances
+    }
+
+    pub fn waypoint(&self) -> Option<Waypoint> {
+        self.waypoint
+    }
+
+    pub fn mint_key_pair(&self) -> &KeyPair<Ed25519PrivateKey, Ed25519PublicKey> {
+        &self.mint_key_pair
+    }
+
+    pub fn validator_and_fullnode_instances(&self) -> impl Iterator<Item = &Instance> {
+        self.validator_instances
+            .iter()
+            .chain(self.fullnode_instances.iter())
+    }
+}
+
+#[derive(Clone)]
+pub struct Instance {
+    peer_name: String,
+    ip: String,
+    json_rpc_port: u32,
+    debug_interface_port: Option<u32>,
+    http_client: Client,
+    backend: InstanceBackend,
```

Comment:
I'm not sure when this needs to hold a "backend"

Reply:
hmmm, this field is needed in somewhere in CT, I hold it there now it may be removed after I dig into actual test where it is been used